### ISSUE TITLE
Consider replacing usages of Window.confirm() with Modal

### ISSUE THREAD

**Issue by noisysocks** [WordPress/gutenberg#16583]
We're currently using [`Window.confirm()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm) in four places. This isn't an ideal experience as we're unable to customise the OK button to have a more descriptive label.

Moving these instances to use a `Modal` component might provide a better experience. Here's an example of a `Modal` being used this way, from @kjellr's mockup in https://github.com/WordPress/gutenberg/issues/16315#issuecomment-507331101:

<img width="464" alt="Screen Shot 2019-07-15 at 16 03 13" src="https://user-images.githubusercontent.com/612155/61197808-27c94180-a71a-11e9-947a-e7f2b14798fc.png">

---

### 1. Switching a published post to draft

Current:

<img width="458" alt="Screen Shot 2019-07-15 at 15 57 17" src="https://user-images.githubusercontent.com/612155/61197709-bc7f6f80-a719-11e9-8089-256826dece35.png">

Proposed:

```
Switch to Draft

Are you sure you want to unpublish this post?

[ Cancel ] [ Unpublish ]
```

### 2. Switching a published post to private

Current:

<img width="458" alt="Screen Shot 2019-07-15 at 15 57 42" src="https://user-images.githubusercontent.com/612155/61197711-bc7f6f80-a719-11e9-97e8-d2e2e397be9d.png">

Proposed:

```
Privately Publish

Are you sure you want to privately publish this post now?

[ Cancel ] [ Privately Publish ]
```

### 3. Deleting a reusable block 

Current:

<img width="460" alt="Screen Shot 2019-07-15 at 15 58 12" src="https://user-images.githubusercontent.com/612155/61197713-bc7f6f80-a719-11e9-8bd6-0dddfc21bf56.png">

Proposed:

```
Delete Reusable Block

Are you sure you want to delete this Reusable Block? 

It will be permanently removed from all posts and pages that use it.

[ Cancel ] [ Delete ]
```

### 4. Pressing _Reset the template_ when template doesn't match post content

Current:

<img width="462" alt="Screen Shot 2019-07-15 at 15 59 21" src="https://user-images.githubusercontent.com/612155/61197715-bd180600-a719-11e9-8f09-083eba5c5d82.png">

Proposed:

```
Reset Template

Resetting the template may result in loss of content, do you want to continue?

[ Cancel ] [ Reset Template ]
```

**Comment by kjellr** [comment]
Do we currently use modals as a stand-in for standard alerts elsewhere? I do find modals like the one pictured to be a little less alarming than the standard browser one, but I'm definitely be curious about any a11y considerations for this.

**Comment by aduth** [comment]
This is technically impossible to implement, at least for the confirmations shown on `beforeunload` (i.e. the unsaved changes prompt when reloading or navigating away). The browser does not give you an opportunity to prevent navigation except by the standard prompts.

**Comment by noisysocks** [comment]
@aduth: None of the `confirm()`s shown above are triggered when reloading or navigating away 🙂

**Comment by aduth** [comment]
> @aduth: None of the `confirm()`s shown above are triggered when reloading or navigating away 🙂

You're totally right! I skimmed the text of the last screenshot and the "result in content loss" had me jump the gun on assuming that's what we were discussing.

For the ones which don't involve navigation, it's totally possible to re-style. I guess it's worth clarifying at least that it _cannot_ be applied for the prompt which asks the user to confirm unsaved changes before navigating away or refreshing a "dirty" post.

**Comment by t-hamano** [comment]
I searched the entire project for this issue with the keyword `window.alert|window.confirm|window.prompt`. Currently, there appears to be only one implementation left, excluding Storybook.

https://github.com/WordPress/gutenberg/blob/0f689f9af5b28ba53f7047a346882296329e13c6/packages/editor/src/components/template-validation-notice/index.js#L18-L22
